packages/shared-ini-file-loader/src/parseIni.ts

```diff
@@ -4,29 +4,32 @@ const profileNameBlockList = ["__proto__", "profile __proto__"];
 
 export const parseIni = (iniData: string): ParsedIniData => {
   const map: ParsedIniData = {};
+
   let currentSection: string | undefined;
+  let currentSubSection: string | undefined;
 
   for (let line of iniData.split(/\r?\n/)) {
     line = line.split(/(^|\s)[;#]/)[0].trim(); // remove comments and trim
     const isSection: boolean = line[0] === "[" && line[line.length - 1] === "]";
     if (isSection) {
+      currentSubSection = undefined;
       currentSection = line.substring(1, line.length - 1);
       if (profileNameBlockList.includes(currentSection)) {
         throw new Error(`Found invalid profile name "${currentSection}"`);
       }
     } else if (currentSection) {
-      const indexOfEqualsSign = line.indexOf("=");
-      const start = 0;
-      const end: number = line.length - 1;
-      const isAssignment: boolean =
-        indexOfEqualsSign !== -1 && indexOfEqualsSign !== start && indexOfEqualsSign !== end;
-      if (isAssignment) {
-        const [name, value]: [string, string] = [
-          line.substring(0, indexOfEqualsSign).trim(),
-          line.substring(indexOfEqualsSign + 1).trim(),
-        ];
-        map[currentSection] = map[currentSection] || {};
-        map[currentSection][name] = value;
+      const [name, value] = line.split("=").map((item) => item.trim());
```

Comment:
can't some tokens contain `=`?

should we keep the split on the first `=`?

Reply:
I'm not sure of examples where values contain `=`.

But since it's currently handled, I fixed it and added a test case [`3d5b7c6` (#986)](https://github.com/awslabs/smithy-typescript/pull/986/commits/3d5b7c69f0e76cdeac4376c3dfa1c8dca845054c)